src/main/java/pages/cartPage/shoppingCartTable/ShoppingCartTableRow.java

```diff
@@ -0,0 +1,199 @@
+package pages.cartPage.shoppingCartTable;
+
+import enums.Currency;
+import lombok.Getter;
+import lombok.Setter;
+import models.ProductModel;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import pages.BasePage;
+import pages.cartPage.BaseTable;
+
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.List;
+
+import static engine.drivers.WebDriverFactory.getWebDriverInstance;
+
+@Getter
+@Setter
+public class ShoppingCartTableRow {
+    public By deleteItemButton = By.xpath(".//i[contains(@class, 'fa-trash-o')]/parent::a");
+    public By quantityItemInput = By.xpath(".//input[contains(@id, 'cart_quantity')]");
+    protected BasePage basePage;
+    protected BaseTable baseTable;
+    protected WebDriver driver = getWebDriverInstance();
+
+    public ShoppingCartTableRow() {
+        this.baseTable = new BaseTable("//div[contains(@class, 'product-list')]/table");
+        this.basePage = baseTable.getBasePage();
+    }
+
+    public ProductModel getProductFromRow(int rowNumber) {
+        var rowContent = getRowContent(rowNumber);
+
+        return ProductModel.builder()
+                .name(rowContent.get(ShoppingCartHeader.NAME.columnIndex))
+                .model(rowContent.get(ShoppingCartHeader.MODEL.columnIndex))
+                .priceCurrency(getCurrencyFromUnitPrice(rowNumber))
+                .price(getUnitPrice(rowNumber))
+                .imageUrl(rowContent.get(ShoppingCartHeader.IMAGE.columnIndex))
+                .build();
+    }
+
+    public List<String> getRowContent(int rowNumber) {
```

Comment:
Do we need that?

Reply:
yes, because row content is not the same as ```ProductModel```. So I first use this method to retrieve the content of the row, and then I use ``public ProductModel getProductFromRow(int rowNumber)`` to create an object of class ``ProductModel`` from that row.

Reply:
**EDIT**
The relationship between ```ShoppingCartTable`` and ```ShoppingCartTableRow`` has been corrected as we discussed. Now the ```ShoppingCartTableRow``` is a class representing a row in the ```ShoppingCartTable``` table. In addition, a ```CheckoutProductModel`` class has been added to represent the product created from this row. Thanks a lot for your help!